frontend/src/utils/wildcard.js

```diff
@@ -0,0 +1,91 @@
+//
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+'use strict'
+
+import startsWith from 'lodash/startsWith'
+import endsWith from 'lodash/endsWith'
+import map from 'lodash/map'
+import filter from 'lodash/filter'
+import head from 'lodash/head'
+import includes from 'lodash/includes'
+
+export function wildcardObjectsFromStrings (wildcardStrings) {
+  return map(wildcardStrings, item => {
+    let startsWithWildcard = false
+    let endsWithWildcard = false
+    let customWildcard = false
+    let value = item
+
+    if (item === '*') {
+      customWildcard = true
+      value = ''
+    } else {
+      if (startsWith(item, '*')) {
+        startsWithWildcard = true
+        value = value.substring(1, value.length)
+      }
+      if (endsWith(item, '*')) {
+        endsWithWildcard = true
+        value = value.substring(0, value.length - 1)
+      }
+    }
+
+    return {
+      value,
+      originalValue: item,
+      startsWithWildcard,
+      endsWithWildcard,
+      customWildcard,
+      isWildcard: startsWithWildcard || endsWithWildcard || customWildcard,
+      test (value) {
+        if (!this.isWildcard) {
+          return value === this.value
+        }
+        if (this.customWildcard) {
+          return true
+        }
+        if (this.startsWithWildcard && this.endsWithWildcard) {
+          return includes(value.substring(1, value.length - 1), this.value)
+        }
+        if (this.startsWithWildcard) {
+          return endsWith(value.substring(1, value.length), this.value)
+        }
+        if (this.endsWithWildcard) {
+          return startsWith(value.substring(0, value.length - 1), this.value)
+        }
```

Comment:
Ok. Now I have understood why you have done it like this. For me the following code would be more intuitive. But I am also fine with the current code if you add some comments. I have have not understood why you remove only one character from start and/or end like it is done for the pattern. 
```suggestion
        const index = value.indexOf(this.value)
        const lengthDifference = value.length - this.value.length

        if (this.startsWithWildcard && this.endsWithWildcard) {
          // included but does not start or end with
          return index > 0 && index < lengthDifference
        }
        if (this.startsWithWildcard) {
          // ends with and not equal
          return index === lengthDifference && lengthDifference > 0
        }
        if (this.endsWithWildcard) {
          // starts with and not equal
          return index === 0 && lengthDifference > 0
        }
```